modules/quality/include/opencv2/quality/QualityBase.hpp

```diff
@@ -0,0 +1,68 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#ifndef OPENCV_QUALITYBASE_HPP
+#define OPENCV_QUALITYBASE_HPP
+
+#include <vector>
+#include <opencv2/core.hpp>
+
+/**
+@defgroup quality Image Quality Analysis (IQA) API
+*/
+
+namespace cv {
+namespace quality {
+namespace detail {
+    /** @brief Type of output quality maps (if generated by the algorithm) */
+    using quality_map_type = UMat;
+}   // detail
+
+//! @addtogroup quality
+//! @{
+
+	/************************************ Quality Base Class ************************************/
+
+	class CV_EXPORTS_W QualityBase
+        : public virtual Algorithm
+	{
+	public:
+
+        /**
+		@brief Destructor
+			*/
+		virtual ~QualityBase() = default;
+
+        /**
+        @brief Compute quality score per channel, see specific algorithm for interpreting result scores
+        @param cmpImgs comparison image(s), can be empty for no-reference quality algorithms
+        */
+        virtual CV_WRAP cv::Scalar compute( InputArrayOfArrays cmpImgs ) = 0;
```

Comment:
What is the meaning of `cv::Scalar` results here?

Reply:
The `cv::Scalar` returns the quality score for each channel in the input image.  For example, if the user provides a BGR image, the scalar returns the quality score for Blue --> scalar[0], Green --> scalar[1], Red --> scalar[2].  Alternatively, if the user only wants to provide a grayscale image, the result will be scalar[0].